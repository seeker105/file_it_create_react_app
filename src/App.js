import React from 'react';
import createHistory from 'history/createBrowserHistory';
import {firebase} from './firebase/firebase';
import {Provider} from 'react-redux';
import store, {persistor} from './store/configureStore';
import AppRouter from './Routers/AppRouter';
import {loginGenerator, logoutGenerator} from './actions/auth';
import {PersistGate} from 'redux-persist/lib/integration/react';
import LoadingPage from './components/LoadingPage';

export const history = createHistory();

const App = () => (
  <div>
    <Provider store={store}>
      <PersistGate loading={<LoadingPage />} persistor={persistor}>
        <AppRouter history={history} />
      </PersistGate>
    </Provider>
  </div>
)

export default App;

const logIn = (firstName, lastName, email) => {
  store.dispatch(loginGenerator(firstName, lastName, email));
  if (history.location.pathname === '/' ||
  history.location.pathname === '/create-account' ||
  history.location.pathname === '/sign-in-form') {
    history.push('/dashboard')
  }
  console.log(store.getState());
}

firebase.auth().onAuthStateChanged((user) => {
  if (user) {
    let firstName, lastName;
    const email = user.email
    if (user.displayName) {
      const names = user.displayName.split(' ');
      firstName = names[0];
      lastName = names[names.length - 1];
      logIn(firstName, lastName, email)
    } else {
      firstName = localStorage.getItem('firstName');
      lastName = localStorage.getItem('lastName');
      user.updateProfile({
        displayName: firstName + ' ' + lastName
      }).then(logIn(firstName, lastName, email))
    }
  } else {
    store.dispatch(logoutGenerator());
    console.log(store.getState());
    history.push('/');
  }
})
