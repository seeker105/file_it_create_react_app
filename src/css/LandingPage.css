.box-layout {
  align-items: center;
  background: url("../images/file-cabinet.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: rgba(244, 244, 244, 0.95);
  border-radius: 5px;
  text-align: center;
  width: 16rem;
  padding: 1rem 1rem 2.5rem 1rem;
}

.box-layout__title {
  line-height: 1;
}
