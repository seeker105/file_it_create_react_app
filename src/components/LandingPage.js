import React from 'react';
import '../css/LandingPage.css';
import {history} from '../App';

class LandingPage extends React.Component {
  onCreateAccountClick = () => {
    history.push("/create-account")
  }

  onSignInWithEmailClick = () => {
    history.push("/sign-in-form")
  }

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">File It!</h1>
          <p>Save your files in the cloud. Download from anywhere.</p>
          <button
            type="button"
            onClick={this.onCreateAccountClick}
            className="button">
            Create Account
          </button>
          <br />
          <button
            type="button"
            onClick={this.onSignInWithEmailClick}
            className="button">
            Sign In
          </button>
        </div>
      </div>
    );
  }
}
export default LandingPage;
