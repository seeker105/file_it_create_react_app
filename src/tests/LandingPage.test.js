import React from 'react';
import LandingPage from '../components/LandingPage';
import {BrowserRouter, Router, MemoryRouter} from 'react-router';
import {shallow, mount} from 'enzyme';
import {render} from 'react-dom';
import {Link, Route} from 'react-router-dom';
import {history} from '../App';
import {Simulate} from 'react-dom/test-utils';

test('should display create account and sign in links', () => {
  const wrapper = mount(<MemoryRouter initialEntries={['/']} initialIndex={0}>
    <LandingPage />
  </MemoryRouter>);
  expect(wrapper.find('a').at(0).text()).toBe("Create Account");
  expect(wrapper.find('a').at(1).text()).toBe("Sign In");
})

// test('experimental tests', () => {
//   const leftClickEvent = {
//     defaultPrevented: false,
//     preventDefault() { this.defaultPrevented = true },
//     metaKey: null,
//     altKey: null,
//     ctrlKey: null,
//     shiftKey: null,
//     button: 0
//   }
//   const div = document.createElement('div');
//   const wrapper2 = render(
//     <MemoryRouter initialEntries={['/']} initialIndex={0}>
//       <LandingPage />
//     </MemoryRouter>, div
//   );

//   console.log(div.innerHTML);
//   Simulate.click(div.querySelector('#landing-page-link-to-create-account-form'), leftClickEvent);
//   console.log(div.innerHTML);
// })
