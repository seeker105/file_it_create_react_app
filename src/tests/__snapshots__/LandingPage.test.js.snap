// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display create account and sign in links 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="box-layout__box"
>
        <h1
                className="box-layout__title"
        >
                File It!
        </h1>
        <p>
                Save your files in the cloud. Download from anywhere.
        </p>
        <button
                className="button"
                onClick={[Function]}
                type="button"
        >
                Create Account
        </button>
        <br />
        <button
                className="button"
                onClick={[Function]}
                type="button"
        >
                Sign In
        </button>
</div>,
      "className": "box-layout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="box-layout__title"
>
            File It!
</h1>,
          <p>
            Save your files in the cloud. Download from anywhere.
</p>,
          <button
            className="button"
            onClick={[Function]}
            type="button"
>
            Create Account
</button>,
          <br />,
          <button
            className="button"
            onClick={[Function]}
            type="button"
>
            Sign In
</button>,
        ],
        "className": "box-layout__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "File It!",
            "className": "box-layout__title",
          },
          "ref": null,
          "rendered": "File It!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save your files in the cloud. Download from anywhere.",
          },
          "ref": null,
          "rendered": "Save your files in the cloud. Download from anywhere.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
            "className": "button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="box-layout__box"
>
          <h1
                    className="box-layout__title"
          >
                    File It!
          </h1>
          <p>
                    Save your files in the cloud. Download from anywhere.
          </p>
          <button
                    className="button"
                    onClick={[Function]}
                    type="button"
          >
                    Create Account
          </button>
          <br />
          <button
                    className="button"
                    onClick={[Function]}
                    type="button"
          >
                    Sign In
          </button>
</div>,
        "className": "box-layout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="box-layout__title"
>
              File It!
</h1>,
            <p>
              Save your files in the cloud. Download from anywhere.
</p>,
            <button
              className="button"
              onClick={[Function]}
              type="button"
>
              Create Account
</button>,
            <br />,
            <button
              className="button"
              onClick={[Function]}
              type="button"
>
              Sign In
</button>,
          ],
          "className": "box-layout__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "File It!",
              "className": "box-layout__title",
            },
            "ref": null,
            "rendered": "File It!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save your files in the cloud. Download from anywhere.",
            },
            "ref": null,
            "rendered": "Save your files in the cloud. Download from anywhere.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
              "className": "button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
